docs/api.md

```diff
@@ -785,6 +785,26 @@ page.evaluate(() => console.log('hello', 5, {foo: 'bar'}));
 
 Emitted when the page crashes. Browser pages might crash if they try to allocate too much memory.
 
+When a page crashes, any ongoing and subsequent operations will throw, so the most common way to deal with crashes is to catch an exception:
+
+```js
+try {
+  await page.click('button');
+} catch (e) {
+  // Exception message might contain 'crash'
+  // indicating that click failed because of a page crash.
+}
+```
+
+However, waiting for an event might stall because of a crash. In this case, handling crash event helps:
```

Comment:
I'd rather recommend `page.waitForEvent` in this case and ensured that it throws if the target crashes.

Reply:
Right, but folks still use events for various reasons. The snippet here is to help those who do use the EventEmitter. Any ideas how to make this better?

Reply:
Fixed.

Reply:
> Right, but folks still use events for various reasons. The snippet here is to help those who do use the EventEmitter. Any ideas how to make this better?

I'd say we should encourage them to use waitForEvent to avoid such a clumsy code (which is what we almost exclusively do ourselves in our tests). With that said I don't know enough about their rationale for using listeners perhaps there is a flaw in the waitForEvent APIs which we should address.